src/hooks/useUserData.ts

```diff
@@ -0,0 +1,12 @@
+import { useGetUserQuery } from '@/app/services/userApi';
+
+const useUserData = () => {
+    const { data } = useGetUserQuery();
+    const adminData = data?.roles.admin;
+    const superUserData = data?.roles.super_user;
+    const isUserAuthorized = !!adminData || !!superUserData;
+    console.log('the data is ', data);
+    return { data, isUserAuthorized };
+};
+
+export default useUserData;
```

Comment:
apart from following naming convention of hooks, it looks like a normal function. do you know how a function & a custom hook differ

Reply:
I think it is a custom hook only. Why do you thick it is not correct?